Apps/WebClient/src/ClientApp/src/components/report/medicationHistory.vue

```diff
@@ -57,20 +57,10 @@ export default class MedicationHistoryReportComponent extends Vue {
 
     private get visibleRecords(): MedicationStatementHistory[] {
         let records = this.medicationStatements.filter((record) => {
-            let filterStart = true;
-            if (this.startDate !== null) {
-                filterStart = new DateWrapper(
-                    record.dispensedDate
-                ).isAfterOrSame(new DateWrapper(this.startDate));
-            }
-
-            let filterEnd = true;
-            if (this.endDate !== null) {
-                filterEnd = new DateWrapper(
-                    record.dispensedDate
-                ).isBeforeOrSame(new DateWrapper(this.endDate));
-            }
-            return filterStart && filterEnd;
+            return (
+                this.filter.allowsDate(record.dispensedDate) &&
```

Comment:
nit: not really a fond of the name "allows" but i'm struggling to suggest something too, just throwing it out there! "shouldDisplayDate", "filterDate" ?!

kudos on the filter refactoring!

Reply:
I thought about it too but that's the best I came up with. I think Should dsiplay date works better, but is a tad verbose while filterDate sounds like it filters dates instead of checking if its valid

Reply:
Maybe `matchesDateFilter()` and `matchesMedicationsFilter()` to mirror `hasDateFilter()` and `hasMedicationsFilter()`?